xla/service/gpu/runtime/dynamic_slice_thunk.h

```diff
@@ -56,7 +56,10 @@ class DynamicSliceThunk : public Thunk {
       std::vector<std::optional<std::vector<Offset>>> offsets,
       std::vector<std::optional<Shape>> orig_shapes,
       std::vector<std::optional<Shape>> sliced_shapes,
-      std::vector<std::optional<uint64_t>> offset_byte_sizes);
+      std::vector<std::optional<uint64_t>> offset_byte_sizes,
+      std::unique_ptr<HloModule> while_init,
```

Comment:
The general approach should be documented as part of this thunk's documentation.